### DISCUSSION TITLE
Allowed Hosts

### DISCUSSION THREAD

**Discussion by cancan101** [symfony/symfony#54427] [Q&A]
Does symfony have a concept of "allowed hosts" (see for example [Django's docs](https://docs.djangoproject.com/en/5.0/ref/settings/#allowed-hosts)) which is:
> A list of strings representing the host/domain names that this Django site can serve. This is a security measure to prevent [HTTP Host header attacks](https://docs.djangoproject.com/en/5.0/topics/security/#host-headers-virtual-hosting), which are possible even under many seemingly-safe web server configurations.

**Comment by Bossman1337** [comment]
You can set the host for your controller routes. [Documentation here](https://symfony.com/doc/7.1/routing.html#sub-domain-routing) with an example.

```
#[Route('/', name: 'index', host: 'example.com')]
    public function index(): Response
{ ...
```
You can also point it to a parameter.
```
#[Route('/', name: 'index', host: '%my.param.name%')]
    public function index(): Response
{ ...
```
If you need a more custom solution, lets say for checking a whitelist of hosts for example, then you can use conditions that use your custom service. [Route matching expressions documentation](https://symfony.com/doc/7.1/routing.html#matching-expressions).

`#[Route(condition: "service('App\\\Service\\\RouteChecker').check(request)")]`

**Reply by cancan101** [reply]
What is the easiest way to do this at a global / server level and not have to set it on each route. For example the Django setting is site wide. 

Perhaps `access_control` with `ROLE_NO_ACCESS` set and a regex for any non matching host name. But that seems clunky.

**Reply by Bossman1337** [reply]
You can create a listener that listens to every request and filter out that way (mentioned below by lyrixx, this is probably the best way tbh). Some [documentation here on listeners and subscribers](https://symfony.com/doc/current/event_dispatcher.html#listeners-or-subscribers)

You could also do route attributes at the class level with a condition check using a service..

```
#[Route(condition: "service('App\\\Service\\\RouteChecker').check(request)")]
class MyController extends AbstractController
{ ...
```

**Comment by lyrixx** [comment]
You can build your own listener and do the logic in it. It's like 4 LOC. It should be simple. 

You must listen kernel.request with a very high priority

**Comment by cancan101** [comment]
I believe the answer is [`trusted_hosts`](https://symfony.com/doc/current/reference/configuration/framework.html#trusted-hosts).
